internal/validations/observability/outputs/cloudwatch.go

```diff
@@ -4,24 +4,72 @@ import (
 	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
 	internalcontext "github.com/openshift/cluster-logging-operator/internal/api/context"
 	"github.com/openshift/cluster-logging-operator/internal/api/observability"
-	"github.com/openshift/cluster-logging-operator/internal/generator/vector/output/cloudwatch"
+	"regexp"
 )
 
 const (
-	RoleARNsOpt       = "roleARNs"
-	ErrInvalidRoleARN = "CloudWatch RoleARN is invalid"
+	ErrInvalidRoleARN       = "CloudWatch RoleARN is invalid"
+	ErrInvalidAssumeRoleARN = "CloudWatch AssumeRole RoleARN is invalid"
 )
 
+// ValidateCloudWatchAuth ensures auth role and assumeRole ARN's are valid
 func ValidateCloudWatchAuth(spec obs.OutputSpec, context internalcontext.ForwarderContext) (results []string) {
 	secrets := observability.Secrets(context.Secrets)
 	authSpec := spec.Cloudwatch.Authentication
 
-	// Validate role ARN
-	if authSpec.Type == obs.CloudwatchAuthTypeIAMRole {
-		roleArn := cloudwatch.ParseRoleArn(authSpec, secrets)
+	switch authSpec.Type {
+	case obs.CloudwatchAuthTypeIAMRole:
+		roleArn := ParseRoleArn(authSpec, secrets)
 		if roleArn == "" {
 			results = append(results, ErrInvalidRoleARN)
 		}
+		if authSpec.IAMRole.AssumeRole != nil {
+			assumeRoleArn := ParseAssumeRoleArn(authSpec, secrets)
+			if assumeRoleArn == "" {
+				results = append(results, ErrInvalidAssumeRoleARN)
+			}
+		}
+	case obs.CloudwatchAuthTypeAccessKey:
+		if authSpec.AWSAccessKey.AssumeRole != nil {
+			assumeRoleArn := ParseAssumeRoleArn(authSpec, secrets)
+			if assumeRoleArn == "" {
+				results = append(results, ErrInvalidAssumeRoleARN)
+			}
+		}
 	}
+
 	return results
 }
+
+// ParseRoleArn search for valid AWS arn
+func ParseRoleArn(auth *obs.CloudwatchAuthentication, secrets observability.Secrets) string {
+	var roleString string
+	if auth.Type == obs.CloudwatchAuthTypeIAMRole {
+		roleString = secrets.AsString(&auth.IAMRole.RoleARN)
+	}
+	return findSubstring(roleString)
+}
+
+// ParseAssumeRoleArn search for valid AWS assumeRole arn
+func ParseAssumeRoleArn(auth *obs.CloudwatchAuthentication, secrets observability.Secrets) string {
+	var roleString string
+	if auth.IAMRole != nil && auth.IAMRole.AssumeRole != nil {
```

Comment:
elsewhere we do a switch on the type.  Any concerns of unintentionally overwriting the value

Reply:
This is called from within a switch, so the type is already determined    The field path should prevent any confusion.   Maybe if both types were present?  a functional test should be able to determine if this works or not.